### ISSUE TITLE
Shouldn't write_msr()'s wrmsr method specify the target cpu?

### ISSUE THREAD

**Issue by cassella** [speed47/spectre-meltdown-checker#294]
The other methods reference the path `/dev/cpu/$_cpu/msr`, but the
wrmsr method passes only the msr number and the value 0 to be written:

```shell
		# for Linux
		# convert to decimal
		if [ ! -w /dev/cpu/"$_cpu"/msr ]; then
			ret=200 # permission error
		# if wrmsr is available, use it
		elif command -v wrmsr >/dev/null 2>&1 && [ "$SMC_NO_WRMSR" != 1 ]; then
			_debug "write_msr: using wrmsr"
			wrmsr $_msr_dec 0 2>/dev/null; ret=$?
		# or if we have perl, use it, any 5.x version will work
		elif command -v perl >/dev/null 2>&1 && [ "$SMC_NO_PERL" != 1 ]; then
			_debug "write_msr: using perl"
			ret=1
			perl -e "open(M,'>','/dev/cpu/$_cpu/msr') and seek(M,$_msr_dec,0) and exit(syswrite(M,pack('H16',0)))"; [ $? -eq 8 ] && ret=0
```

The wrmsr command has a `-p` option:

```
$ wrmsr -h
Usage: wrmsr [options] regno value...
  --help         -h  Print this help
  --version      -V  Print current version
  --all          -a  all processors
  --processor #  -p  Select processor number (default 0)
```

(Also, the "convert to decimal" comment seems stale or misplaced.)

**Comment by speed47** [comment]
Thanks, this one was long overdue! Fixed in the [test version](https://raw.githubusercontent.com/speed47/spectre-meltdown-checker/refs/heads/test-build/spectre-meltdown-checker.sh).

.

**Comment by speed47** [comment]
Released https://github.com/speed47/spectre-meltdown-checker/releases
